**Antonio Foglia, af3152**

Notebook: https://github.com/Antonio-Foglia/ASTRC3105

In [36]:
import numpy as np
import math
import astropy as ap
from astropy import constants as const
from astropy  import units as u

# Q1

## a)

In [37]:
def a(P, M_star):
    return ((P**2 * const.G * M_star / (4 * np.pi**2))**(1/3)).decompose()

a(100*u.day,0.6*const.M_sun)

<Quantity 5.31998259e+10 m>

In [38]:
def transit_prob(period, M_star, R_star):
    Prob = 0.005*(R_star / const.R_sun) * (a(period, M_star) / const.au)**(-1)
    return Prob.decompose()

In [39]:
transit_prob(100 * u.day, 0.6*const.M_sun, 0.51*const.R_sun)

<Quantity 0.0071706>

## b)

In [40]:
def tau(period, M_star, R_star):
    t = 13*(M_star / const.M_sun)**(-1/2) * (a(period, M_star) / const.au)**(1/2) * (R_star / const.R_sun) * u.hour
    return t.to(u.hour)

In [41]:
tau(100 * u.day, 0.6*const.M_sun, 0.51*const.R_sun)

<Quantity 5.10423079 h>

## c)

In [42]:
Rp = 3389.5 * 1000 * u.m

In [43]:
(Rp/(0.51*const.R_sun)).decompose()

<Quantity 0.00955308>

# Q2

In [44]:
M = 0.69*const.M_sun
R = 0.64*const.R_sun

def planet_R(min_rel_flux):
    flux_change = (1 - min_rel_flux)
    # print(f"DL/L*: {flux_change}\n")
    
    Rp = R * np.sqrt(flux_change)
    #print(f"Rp: {Rp.to(u.R_earth)}\n")
    return Rp.to(u.R_earth)

def orbital_R(duration):
    a = const.au * (duration/13)**2 * (M/const.M_sun) * (R/const.R_sun)**(-2) * (1/u.hour)**(2)
    #print(f"a: {a.to(u.AU)}\n")
    return a.to(u.AU)

## Kepler-62b

In [45]:
print(f'a) R_p = {planet_R(0.9996).round(3)}\nb) a = {orbital_R(2.9 * u.hour).round(3)}')

a) R_p = 1.396 earthRad
b) a = 0.084 AU


## Kepler-62c

In [46]:
print(f'a) R_p = {planet_R(0.9999).round(3)}\nb) a = {orbital_R(3.5 * u.hour).round(3)}')

a) R_p = 0.698 earthRad
b) a = 0.122 AU


## Kepler-62d

In [47]:
print(f'a) R_p = {planet_R(0.9990).round(3)}\nb) a = {orbital_R(3.5 * u.hour).round(3)}')

a) R_p = 2.208 earthRad
b) a = 0.122 AU


## Kepler-62e

In [48]:
print(f'a) R_p = {planet_R(0.9993).round(3)}\nb) a = {orbital_R(7.1 * u.hour).round(3)}')

a) R_p = 1.847 earthRad
b) a = 0.502 AU


## Kepler-62f

In [49]:
print(f'a) R_p = {planet_R(0.9995).round(3)}\nb) a = {orbital_R(8 * u.hour).round(3)}')

a) R_p = 1.561 earthRad
b) a = 0.638 AU


## c)

In [50]:
a(18.2 * u.day, M).to(u.AU)

<Quantity 0.11965523 AU>

The above is quite similar to what I got in the calculation in part *b)*: $0.122$ AU.

We were making the assumption that the orbits were perfectly circular, which is often not the case. Moreover we were assuming $i \approx 90^{\circ}$ which means that the planet is orbiting on the same plane as the observer, which is also unlikely.